guides/creating-a-custom-form-field-control.md

```diff
@@ -326,15 +326,51 @@ private _disabled = false;
 
 #### `errorState`
 
-This property indicates whether the associated `NgControl` is in an error state. In this example,
-we show an error if the input is invalid and our component has been touched.
+This property indicates whether the associated `NgControl` is in an error state. For example,
+we can show an error if the input is invalid and our component has been touched.
 
 ```ts
 get errorState(): boolean {
   return this.parts.invalid && this.touched;
 }
 ```
 
+However, there are some error triggers that we can't subscribe to (e.g. parent form submissions),
+to handle such cases we should re-evaluate `errorState` on every change detection cycle.
+
+```ts
+/** Whether the component is in an error state. */
+errorState: boolean = false;
+
+constructor(
+  ...,
+  @Optional() private _parentForm: NgForm,
+  @Optional() private _parentFormGroup: FormGroupDirective
+) {
+...
+}
+
+ngDoCheck() {
+  if (this.ngControl) {
```

Comment:
Could we not subscribe to `(this._parentFormGroup || this.parentForm).ngSubmit` instead of doing it in `doCheck`?

Reply:
based on the comment in [matInput](https://github.com/angular/components/blob/main/src/material/input/input.ts#L353C4-L353C4), form submission is just an example of a error triggers that we can't subscribe to. I'm not sure, but maybe there are other error triggers that we can't subscribe and it's not just about submission.

Reply:
I guess that's fair. I think the submitted state is the only thing for the default ErrorStateMatcher, but someone could define a custom one that depends on other things